src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +109,845 @@ public static void init()
-     * @throws IOException if nothing found.
+     * @param prefix   either schematics, scans, cache
      */
-    public static void loadStyleMaps(final Path basePath) throws IOException
+    private static void loadSchematicsForPrefix(@NotNull final Path basePath, @NotNull final String prefix)
     {
-        try (Stream<Path> walk = Files.walk(basePath))
+        Log.getLogger().info("Structures: loadSchematicsForPrefix(" + basePath + ", " + prefix + ")");
+        try (Stream<Path> walk = Files.walk(basePath.resolve(prefix)))
         {
             final Iterator<Path> it = walk.iterator();
 
             while (it.hasNext())
             {
                 final Path path = it.next();
 
-                final String style = path.getParent().getFileName().toString();
-
-                //Don't treat generic schematics as decorations or huts - ex: supply ship
-                if (NULL_STYLE.equals(style) || MINER_STYLE.equals(style))
-                {
-                    continue;
-                }
-
                 if (path.toString().endsWith(SCHEMATIC_EXTENSION))
                 {
-                    final String filename = path.getFileName().toString().split("\\.nbt")[0];
-                    final String hut = filename.split("\\d+")[0];
+                    String relativePath = path.toString().substring(basePath.toString().length()).split("\\" + SCHEMATIC_EXTENSION)[0];
+                    if (!path.getFileSystem().getSeparator().equals("/"))
+                    {
+                        relativePath = relativePath.replace(path.getFileSystem().getSeparator(), "/");
+                    }
+                    if (relativePath.startsWith("/"))
+                    {
+                        relativePath = relativePath.substring(1);
+                    }
 
-                    if (isSchematicHut(hut))
+
+
+                    final StructureName structureName = new StructureName(relativePath);
+                    final String md5 = Structure.calculateMD5(Structure.getStream(relativePath));
+                    if (md5 == null)
                     {
-                        addHutStyle(hut, style);
-                        incrementHutMaxLevel(hut);
+                        Log.getLogger().error("Structures: " + structureName + " with md5 null.");
                     }
                     else
                     {
-                        addDecorationStyle(style, filename);
+
+                        if (md5Map.containsKey(structureName.toString()))
+                        {
+                            Log.getLogger().info("Override " + structureName + " md5:" + md5);
+                        }
+                        else
+                        {
+                            Log.getLogger().info("Add " + structureName + " md5:" + md5);
+                        }
+
+                        md5Map.put(structureName.toString(), md5);
+                        final int MAX_TOTAL_SIZE = 32767;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Add underscores to this numeric value for readability [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2148)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)